shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java

```diff
@@ -87,6 +94,54 @@ public boolean hasRightOnStudy(Long studyId, String rightStr) {
         return commService.hasRightOnStudy(studyId, rightStr);
     }
     
+    /**
+	 * Check that the connected user has the given right for the given subject.
+	 * 
+	 * @param studyId the study id
+	 * @param rightStr the right
+	 * @return true or false
+	 */
+    public boolean hasRightOnSubjectId(Long subjectId, String rightStr) {
+    	if (KeycloakUtil.getTokenRoles().contains(ROLE_ADMIN)) {
+			return true;
+		}
+    	Optional<Subject> subject = subjectRepository.findById(subjectId);
+    	if (subject.isEmpty()) {
+    		return false;
+    	}
+    	for (SubjectStudy subjectStudy : subject.get().getSubjectStudyList()) {
+    		boolean hasRight = commService.hasRightOnStudy(subjectStudy.getStudy().getId(), rightStr);
+    		if (hasRight) {
+    			return true;
+    		}
+    	}
+    	return false;
+    }
+    
+    /**
+	 * Check that the connected user has the given right for the given subject.
+	 * 
+	 * @param subjectName the study name
+	 * @param rightStr the right
+	 * @return true or false
+	 */
+    public boolean hasRightOnSubjectName(String subjectName, String rightStr) {
```

Comment:
Maybe refactor the two methods, load subject by subject name then call only one method with subject in entry.

Reply:
Actually, hasRightOnSubjectId is not used no ?